pkg/proxy/topology_test.go

```diff
@@ -476,3 +476,138 @@ func TestFilterTopologyEndpoint(t *testing.T) {
+			// endpoints
+			Name:                  "no internalTrafficPolicy with empty endpoints",
+			internalTrafficPolicy: nil,
+			endpoints:             []endpoint{},
+			expected:              []endpoint{},
+		},
+		{
+			// Case[1]: no internalTrafficPolicy set with non-empty
+			// endpoints
+			Name:                  "no internalTrafficPolicy with non-empty endpoints",
+			internalTrafficPolicy: nil,
+			endpoints: []endpoint{
+				{"10.0.0.0:80", true},
+				{"10.0.0.1:80", false},
+			},
+			expected: []endpoint{
+				{"10.0.0.0:80", true},
+				{"10.0.0.1:80", false},
+			},
+		},
+		{
+			// Case[2]: internalTrafficPolicy is All
+			Name:                  "internalTrafficPolicy is all",
+			internalTrafficPolicy: &all,
+			endpoints: []endpoint{
+				{"10.0.0.0:80", true},
+				{"10.0.0.1:80", false},
+			},
+			expected: []endpoint{
+				{"10.0.0.0:80", true},
+				{"10.0.0.1:80", false},
+			},
+		},
+		{
+			// Case[3]: internalTrafficPolicy is local and there are
+			// non-zero local endpoints
+			Name:                  "internalTrafficPolicy is local with non-zero local endpoints",
+			internalTrafficPolicy: &local,
+			endpoints: []endpoint{
+				{"10.0.0.0:80", true},
+				{"10.0.0.1:80", false},
+			},
+			expected: []endpoint{
+				{"10.0.0.0:80", true},
+			},
+		},
+		{
+			// Case[4]: internalTrafficPolicy is local and there are
+			// zero local endpoints
+			Name:                  "internalTrafficPolicy is local with zero local endpoints",
+			internalTrafficPolicy: &local,
+			endpoints: []endpoint{
+				{"10.0.0.0:80", false},
+				{"10.0.0.1:80", false},
+				{"10.0.0.2:80", false},
+			},
+			expected: []endpoint{},
+		},
+		{
+			// Case[5]: internalTrafficPolicy is preferLocal and there are
+			// zero local endpoints
+			Name:                  "internalTrafficPolicy is preferLocal with zero local endpoints",
+			internalTrafficPolicy: &preferLocal,
+			endpoints: []endpoint{
+				{"10.0.0.0:80", false},
+				{"10.0.0.1:80", false},
+			},
+			expected: []endpoint{
+				{"10.0.0.0:80", false},
+				{"10.0.0.1:80", false},
+			},
+		},
+		{
+			// Case[6]: internalTrafficPolicy is preferLocal and there are
+			// non-zero local endpoints
+			Name:                  "internalTrafficPolicy is preferLocal with zero local endpoints",
+			internalTrafficPolicy: &preferLocal,
+			endpoints: []endpoint{
+				{"10.0.0.0:80", false},
+				{"10.0.0.1:80", false},
+				{"10.0.0.2:80", true},
+				{"10.0.0.3:80", true},
+				{"10.0.0.4:80", true},
+			},
+			expected: []endpoint{
+				{"10.0.0.2:80", true},
+				{"10.0.0.3:80", true},
+				{"10.0.0.4:80", true},
+			},
+		},
+	}
+	endpointsToStringArray := func(endpoints []endpoint) []string {
+		result := make([]string, 0, len(endpoints))
+		for _, ep := range endpoints {
+			result = append(result, ep.Endpoint)
+		}
+		return result
+	}
+	for _, tc := range testCases {
+		t.Run(tc.Name, func(t *testing.T) {
+			m := make(map[Endpoint]endpoint)
+			endpoints := []Endpoint{}
```

Comment:
Would it be cleaner to just have `[]Endpoint{}` as both the input and expected output in the test table?

Reply:
This way we don't have to copy into another map